jina/executors/encoders/multimodal/__init__.py

```diff
@@ -0,0 +1,37 @@
+__copyright__ = "Copyright (c) 2020 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+from typing import Dict
+
+import numpy as np
+
+from jina.executors.decorators import batching, as_ndarray
+from ... import BaseExecutor
+
+
+class BaseMultiModalEncoder(BaseExecutor):
+    """
+    :class:`BaseMultiModalEncoder` encodes data from multiple inputs (``text``, ``buffer``, ``blob`` or other ``embeddings``)
+    into a single ``embedding``
+    """
+
+    def __init__(self,
+                 position_by_modality: Dict[str, int] = {},
+                 *args,
+                 **kwargs):
+        """
+        :param position_by_modality: the map of arguments indicating in which order the modalities they need to come
+        for the encoding method
+        :return:
+        """
+        super().__init__(*args, **kwargs)
+        self.position_by_modality = position_by_modality
```

Comment:
is `self.position_by_modality` this really required at executor level? not seeing the executor using it, it is just used to inform the driver "please order in this way and give it to me". Perhaps just put `position_by_modality` directly into the driver?

Reply:
It is not really required, I thought that the executor should have more knowledge than the driver on what is expected, but both are valid options.

Reply:
@bwanglzu I had a better thought and yes it makes more sense to have it at driver level. It is a quick change

Reply:
@bwanglzu second thoughts. I would keep it as attribute of executor and access via property in the driver. This way the executor can use that to interpret the arguments of the encoders.

Reply:
What about grouping them into `dict` and passing into the executors?

Reply:
It makes a much worse interface. But also in my opinion mulimodal is about having N inputs that come from different  sources and that need N different treatments. It seems a fair assumption in my opinion